presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1679,6 +1680,40 @@ protected Scope visitExcept(Except node, Optional<Scope> scope)
             return visitSetOperation(node, scope);
         }
 
+        private boolean onConditionReferencesRelatedFields(Expression expression, Relation relation)
```

Comment:
you mean `isConditionReferencesRelatedFields`?

Reply:
I meant to say JOIN ON condition....
maybe it is better to name it `isJoinOnConditionReferencesRelatedFields`?